src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -43,114 +43,128 @@ class SvPileup
-      val coalesced = breakpointsIter.takeWhile(_._1.sameBreakEnds(first)).toIndexedSeq
-      val countsMap: Map[EvidenceType, Long] = coalesced
-        .groupBy(_._1.evidence)
-        .map { case (svType: EvidenceType, data: Seq[(PutativeBreakpoint, Long)]) => (svType, data.map(_._2).sum) }
-      val leftRefName  = dict(first.leftRefIndex).name
-      val rightRefName = dict(first.rightRefIndex).name
-      val counts       = EvidenceType.values.map(svType => countsMap.getOrElse(svType, 0L))
+
+    val breakpoints = tracker.breakpoints
+      .toIndexedSeq
+      .sortWith(Breakpoint.PairedOrdering.lt)
+
+    breakpoints.foreach { bp =>
+      val id           = tracker.idOf(bp)
+      val counts       = tracker.countsFor(bp)
+      val leftRefName  = dict(bp.leftRefIndex).name
+      val rightRefName = dict(bp.rightRefIndex).name
       val total        = counts.sum
-      // FIXME: wouldn't it be nice if this wasn't a comma-delimited list?  That would take a bunch of work to use
-      // .sameBreakends and store evidence counts when initially gathering the breakpoints.  Defer to later.
-      val id = coalesced.iterator.map(_._1.id).toSet.toSeq.sorted.mkString(",")
-      writer.write(f"$id\t$leftRefName\t${first.leftPos}\t$rightRefName\t${first.rightPos}\t${first.sameStrand}\t$total")
-      counts.foreach(count => writer.write(f"\t$count"))
-      writer.write('\n')
+
+      val values = Iterator(
+        id, leftRefName, bp.leftPos, toStrand(bp.leftPositive), rightRefName, bp.rightPos, toStrand(bp.rightPositive), total
+      ) ++ counts.iterator
+      writer.write(values.mkString("", "\t", "\n"))
     }
+
     writer.close()
   }
 
+  /** Converts the boolean strand info to +/-. */
+  private def toStrand(positive: Boolean): String = if (positive) "+" else "-"
+
   /** Returns true if the primary alignment is mapped and has sufficient mapping quality */
   private def primaryOk(primary: SamRecord): Boolean = {
     primary.mapped && primary.mapq >= minPrimaryMappingQuality
   }
 }
 
 object SvPileup extends LazyLogging {
-
   val SamEvidenceTag: String = "ev"
   val SamBreakpointTag: String = "be"
 
-  /** A little class to map a breakpoint to id, ignoring the existing id of the breakpoint. */
-  private class BreakpointToIdMap {
-    private var _nextId: Long = 0L
-    private val breakpointToId = mutable.HashMap[PutativeBreakpoint, Long]()
-
-    /** Returns the identifier of the breakpoint, if it exists, or the next identifier that should be used. */
-    def apply(breakpoint: PutativeBreakpoint): Long = {
-      // NB: use id=0 and evidence=any so we can find it in the map regardless of existing id and evidence type
-      this.breakpointToId.getOrElseUpdate(breakpoint.copy(id=0, evidence=EvidenceType.ReadPairTandem), {
-        val id = this._nextId
-        this._nextId += 1
-        id
-      })
+  /** Class that tracks IDs and counts for Breakpoints during discovery. */
+  private class BreakpointTracker {
```

Comment:
I wonder if this is the simplest interface, given method below like `breakpoints`, `countsFor` and `evidenceFor` are present.  You could also implement `Map[Breakpoint, ...]` but you have two maps.  So perhaps it's fine the way it is.  Something to think about.
```suggestion
  private class BreakpointTracker extends Iterator[(Breakpoint, SimpleCounter[EvidenceType]] {
```

Reply:
It's possible there's a better formulation.  But for now the interface is pretty narrow, and it's private, so it'll be easy to change.